### ISSUE TITLE
Support building on Ubuntu 20.04

### ISSUE THREAD

**Issue by sambercovici** [VectorCamp/vectorscan#180]
Splitting from #176 
Currently Vectorscan does not compile on ubuntu 20.04.
To manage that I patched src/util/arch/arm/cpuid_inline.h and added #include <asm/hwcap.h> just before include <sys/auxv.h>

Building with only Neon support then works.

Building with SVE did not work (ISSUE-3 in #176), is it supposed to be supported on ubuntu 20.04 with the clang release it has?

IF SVE is supported, can FAT_RUNTIME be modified so if I turn off SVE2, it will only try to choose between Neon and SVE?
In other words the following:
cmake .. -DBOOST_ROOT=/home/ubuntu/src/vectorscan/build/boost_1_57_0/ -DCMAKE_BUILD_TYPE=Release -DFAT_RUNTIME=on -DBUILD_SVE2_BITPERM=off -DBUILD_SVE2=off -DBUILD_SVE=on

AFIK, in Hyperscan FAT_RUNTIME=on will not include disabled accelerations

**Comment by markos** [comment]
We may add 20.04 support for Neon only. SVE-only support will not be added for 20.04, mostly because of old compiler support and because most of the optimizations are mainly for SVE2. SVE has only a few basic instructions that can be used as it's mostly for FP operations.

**Comment by sambercovici** [comment]
Will the 20.04 support be done in the same way I did or differently?
When is 5.4.11 expected?

**Comment by markos** [comment]
It will do something along those lines, yes. As I already said in the other bug report, next release should be done before the end of October.

**Comment by sambercovici** [comment]
Amazing work!

**Comment by markos** [comment]
@sambercovici could you please checkout and test PR #186 on 20.04 and see if it solves your problem? So that I can merge this. I don't have the a 20.04 installation and as I said we don't plan to support it specifically, but I would like you help your usecase.

**Comment by markos** [comment]
With #186 and #188 merged this is considered fixed.
